xbmc/input/touch/generic/GenericTouchActionHandler.cpp

```diff
@@ -161,43 +166,28 @@ int CGenericTouchActionHandler::QuerySupportedGestures(float x, float y)
   return result;
 }
 
-void CGenericTouchActionHandler::touch(uint8_t type, uint8_t button, uint16_t x, uint16_t y)
-{
-  XBMC_Event newEvent;
-  memset(&newEvent, 0, sizeof(newEvent));
-  
-  newEvent.type = type;
-  newEvent.button.button = button;
-  newEvent.button.x = x;
-  newEvent.button.y = y;
-  
-  CWinEvents::MessagePush(&newEvent);
-}
-
-void CGenericTouchActionHandler::sendEvent(int actionId, float x, float y, float x2 /* = 0.0f */, float y2 /* = 0.0f */, int pointers /* = 1 */)
+void CGenericTouchActionHandler::sendEvent(int actionId, float x, float y, float x2 /* = 0.0f */, float y2 /* = 0.0f */, float x3, float y3, int pointers /* = 1 */)
 {
-  XBMC_Event newEvent;
-  memset(&newEvent, 0, sizeof(newEvent));
+  XBMC_Event newEvent{XBMC_TOUCH};
   
-  newEvent.type = XBMC_TOUCH;
   newEvent.touch.action = actionId;
   newEvent.touch.x = x;
   newEvent.touch.y = y;
   newEvent.touch.x2 = x2;
   newEvent.touch.y2 = y2;
+  newEvent.touch.x3 = x3;
+  newEvent.touch.y3 = y3;
   newEvent.touch.pointers = pointers;
 
   CWinEvents::MessagePush(&newEvent);
 }
 
 void CGenericTouchActionHandler::focusControl(float x, float y)
 {
-  XBMC_Event newEvent;
-  memset(&newEvent, 0, sizeof(newEvent));
+  XBMC_Event newEvent{XBMC_SETFOCUS};
 
-  newEvent.type = XBMC_SETFOCUS;
-  newEvent.focus.x = (uint16_t)x;
-  newEvent.focus.y = (uint16_t)y;
+  newEvent.focus.x = static_cast<int> (std::round(x));
```

Comment:
XBMC_SetFocusEvent has `int`s, why cut it to 16 bit? https://github.com/xbmc/xbmc/blob/master/xbmc/windowing/XBMC_events.h#L106